src/cbc_sdk/platform/devices.py

```diff
@@ -220,6 +255,104 @@ def get_vulnerabilties(self):
         return VulnerabilityQuery(Vulnerability, self._cb, self)
 
 
+class DeviceFacet(UnrefreshableModel):
+    """Represents a device field in a facet search."""
+    urlobject = "/appservices/v6/orgs/{0}/devices/_facet"
+    primary_key = "id"
+    swagger_meta_file = "platform/models/device_facet.yaml"
+
+    def __init__(self, cb, model_unique_id, initial_data=None):
+        """
+        Initialize the DeviceFacet object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            model_unique_id (str): Not used.
+            initial_data (dict): Initial data used to populate the facet.
+        """
+        super(DeviceFacet, self).__init__(cb, model_unique_id, initial_data, force_init=False, full_doc=True)
+        self._values = [DeviceFacet.DeviceFacetValue(cb, self, item['id'], item) for item in initial_data['values']]
+
+    class DeviceFacetValue(UnrefreshableModel):
+        """Represents a value of a particular field."""
+        def __init__(self, cb, outer, model_unique_id, initial_data):
+            """
+            Initialize the DeviceFacetValue object.
+
+            Args:
+                cb (BaseAPI): Reference to API object used to communicate with the server.
+                outer (DeviceFacet): Reference to outer facet object.
+                model_unique_id (str): Value ID.
+                initial_data (dict): Initial data used to populate the facet value.
+            """
+            super(DeviceFacet.DeviceFacetValue, self).__init__(cb, model_unique_id, initial_data, force_init=False,
+                                                               full_doc=True)
+            self._outer = outer
+
+        def query_devices(self):
+            """
+            Set up a device query to find all devices that match this facet value.
+
+            Example:
+                >>> facets = api.select(Device).where('').facets(['policy_id'])
+                >>> for value in facets[0].values_:
+                ...     print(f"Policy ID = {value.id}:")
+                ...     for dev in value.query_devices():
+                ...         print(f"    {dev.name} ({dev.last_external_ip_address})")
+
+            Returns:
+                DeviceQuery: A new DeviceQuery set with the criteria, which may have additional criteria added to it.
+            """
+            query = self._cb.select(Device)
+            if self._outer.field == 'policy_id':
+                query.set_policy_ids([int(self.id)])
+            elif self._outer.field == 'status':
+                query.set_status([self.id])
+            elif self._outer.field == 'os':
+                query.set_os([self.id.upper()])
+            elif self._outer.field == 'ad_group_id':
+                query.set_ad_group_ids([int(self.id)])
```

Comment:
Can we add a catch all here if the backend supports additional facet options in the future that way this code doesn't need updated every time a new property is supported? We should be able to check the term name and add it to the criteria object directly using `_update_criteria`

Reply:
This would make sense, but there's not a one-to-one correspondence between the field names as used by faceting and the field names in the criteria.  Some map straight across (`status` => `status`, `os` => `os`), some map to their plurals (`policy_id` => `policy_ids`, `ad_group_id` => `ad_group_ids`).

Reply:
Also, there needs to be conversion done on some fields.  The value as reported in the facet is always a string, but `policy_ids` and `ad_group_ids` require integers.  And `os` is reported by faceting as lower case, but the criteria for it demands upper case.

Reply:
Fair point. I did not account for the change in property names